src/rules/selector-max-specificity/index.js

```diff
@@ -35,15 +35,21 @@ export default function (max) {
         resolvedNestedSelector(selector, rule).forEach(resolvedSelector => {
           // Return early if selector contains a not pseudo-class
           if (selector.indexOf(":not(") !== -1) { return }
+          // Return early if selector contains a matches
+          if (selector.indexOf(":matches(") !== -1) { return }
```

Comment:
> won't a PostCSS warning be produced for each :not and :matches pseudo-classes in a source when this rule is turned off?

I'm not sure I'm following: Why would this be the case? If this rule is turned off, we should never enter the code in this file, right?